StoreApp/Controllers/StoreController.cs

```diff
@@ -39,6 +43,34 @@ public ActionResult Index()
             }
         }
 
+        [HttpPost]
+        public ActionResult Index(FilterProductsViewModel filter)
+        {
+            TempData["Filter"] = filter;
+
+            if (!ModelState.IsValid)
+            {
+                var products = productService.GetAll().Select(p => webMapper.Map(p));
+
+                return View(products);
+            }
+            
+            FilterProductsDTO filterDTO = webMapper.config.Map<FilterProductsViewModel, FilterProductsDTO>(filter);
+            
+            var productsFiltered = filterProductsService.GetProductsDTOFiltered(filterDTO).Select(p => webMapper.Map(p));
+
+            return View(productsFiltered);
+        }
+
+        public ActionResult FilterMenuPartial()
```

Comment:
If you mark all methods by using attributes, don't skip this one

Reply:
fixed